File: 2016-2017/MP/Pracownia 4/DanielDubielTests.hs
-- Wymagamy, by moduł zawierał tylko bezpieczne funkcje
{-# LANGUAGE Safe #-}
-- Definiujemy moduł zawierający testy.
-- Należy zmienić nazwę modułu na {Imie}{Nazwisko}Tests gdzie za {Imie}
-- i {Nazwisko} należy podstawić odpowiednio swoje imię i nazwisko
-- zaczynające się wielką literą oraz bez znaków diakrytycznych.
module DanielDubielTests(tests) where

-- Importujemy moduł zawierający typy danych potrzebne w zadaniu
import DataTypes

-- Lista testów do zadania
-- Należy uzupełnić jej definicję swoimi testami
tests :: [Test]
tests =
  [ Test "val_inc"              (SrcString "input y in y")                                                  (Eval [5] (Value 5))
  , Test "val_inc1"             (SrcString "input x in x + 1")                                              (Eval [42] (Value 43))
  , Test "val_unarny"           (SrcString "input x in - x")                                                (Eval [5] (Value (-5)))
  , Test "val_letOP5priority"   (SrcString "let x = 5 in x + 6 - x + x")                                    (Eval [] (Value 11))
  , Test "val_letOP6priority"   (SrcString "let x = 5 in let y = 6 in x + y * x * x div y")                 (Eval [] (Value 30))
  , Test "val_if"               (SrcString "input x in if x>5 then 3 else 6")                               (Eval [10] (Value 3))
  , Test "val_if2"              (SrcString "input x y in if x>y or x<y then 1 else 0")                      (Eval [5,5] (Value 0))
  , Test "val_if3"              (SrcString "input x y z in if x>y or x<y and z = 5 then 1 else 0")          (Eval [4,6,5] (Value 1))
  , Test "val_ifNeg"            (SrcString "input a b in if not a = b then 1 else 0")                       (Eval [4,6] (Value 1))
  , Test "val_let"              (SrcString "input x in let x = x + 5 in let x = x * x in let x = x+1 in x") (Eval [5] (Value 101))
  , Test "val_contr"            (SrcString "let x = - - 5 in x")                                            (Eval [] (Value 5))
  , Test "val_combined"         (SrcString "input x in let x = 5 in if x = 6 and x <=6  then 10 else 0")    (Eval [10] (Value 0))
  , Test "type_TandF"           (SrcString "input u in true")                TypeError
  , Test "type_mod"             (SrcString "true mod (not false)")                   TypeError
  , Test "type_badReturn"       (SrcString "input x in false")                     TypeError
  , Test "type_undefVar"        (SrcString "x")                                    TypeError
  , Test "type_let"             (SrcString "let x = 6 in true")                    TypeError
  , Test "type_let2"            (SrcString "let x = not 6 in 5")                   TypeError
  , Test "type_if"              (SrcString "if 2 + 2 then 3 else 6")               TypeError
  , Test "type_if2"             (SrcString "if true or 2 + 2 then 3 else 6")       TypeError
  , Test "type_ifNeg"           (SrcString "input x in if not x then 1 else 2")    TypeError
  , Test "type_add"             (SrcString "input x in x + true")                  TypeError
  ]

File: 2016-2017/MP/Pracownia 3/daniel_dubiel_tests.pl
% Definiujemy moduł zawierający testy.
% Należy zmienić nazwę modułu na {imie}_{nazwisko}_tests gdzie za
% {imie} i {nazwisko} należy podstawić odpowiednio swoje imię
% i nazwisko bez wielkich liter oraz znaków diakrytycznych
:- module(daniel_dubiel_tests, [tests/3]).

% Zbiór faktów definiujących testy
% Należy zdefiniować swoje testy
tests(empty_program, input(""), program([])).

tests(easyInvalid1, input("def main()"), no).
tests(easyInvalid2, input("
def main(A) =  
	if A = 0 
	then ((1)
"), no).

tests(mediumInvalid1, file('prog2.hdml'), no).

tests(easy1, input("
def main(A) = 
	A"), yes).
tests(easy2, input("def main(A) = A*A"), yes).
tests(easy3, input("def main(A) = A,A&A"), yes).
tests(easy4, input("
def main(A,B) = 
	if A*B>B 
	then A,A&A
	else B|A,B
"), yes).

tests(med2, input("
def _minor((A,B),_) =
 if 
	if 
		if []
        then ~69 
        else ~~~~42
    then A & 19 > yenefer #[]
    else let [] = [][[]] in 84[[] .. Jaskier]
 then A=B
 else A[B]
"), yes).

tests(medium1, file('prog1.hdml'), yes).
tests(medium2, file('prog3.hdml'), yes).
tests(medium3, file('prog4.hdml'), no).


tests(skos1, input("def main(_) = 1"),
	program([def(main, wildcard(file(test, 1, 10, 9, 1)), num(no, 1))])).
tests(skos2, file('skos.hdml'), yes).

